cpp/posix.h

```diff
@@ -81,11 +81,15 @@ inline int close(int fd) {
 }
 
 inline int putenv(Str* name, Str* value) {
-  int env_string_size =
-      name->len_ + value->len_ + 1;  // NOTE(Jesse): +1 for the '=' between them
-  char* env_string = static_cast<char*>(malloc(env_string_size));
-  snprintf(env_string, env_string_size, "%s=%s", name->data_, value->data_);
-  return ::putenv(env_string);
+  // NOTE(Jesse): Technically this over-allocates by 1 byte, but by doing that
+  // we don't rely on the underlying Str* data_ buffer being defined as (Str::len_ + 1)
```

Comment:
Hm I don't see what the problem was?  What was the bug that is fixed?

If so it's useful to demonstrate the bug in `cpp/leaky_bindings_test.cc`

I think you should be able to rely on the NUL terminator.  The length of the string is N and the length of the allocated buffer is N+1.  This should be true in both leaky and GC cases.

Reply:
I think the comment is confusing me.  If it was truncating before, then why is this over-allocating by one byte?

Again I would favor the test rather than a comment